openmetadata-ui/src/main/resources/ui/src/components/MyData/FeedWidget/FeedWidget.component.tsx

```diff
@@ -0,0 +1,289 @@
+    entityThread: feedList,
+    getFeedData,
+  } = useActivityFeedProvider();
+
+  useEffect(() => {
+    if (feedList) {
+      const { updatedFeedList } = getFeedListWithRelativeDays(feedList);
+      setEntityThread(updatedFeedList.slice(0, 8)); // Limit to 8 items for widget
+    }
+  }, [feedList]);
+
+  // Fetch feed data when component mounts or filter changes
+  useEffect(() => {
+    if (currentUser && getFeedData) {
+      getFeedData(
+        selectedFilter, // filterType
+        undefined, // after
+        ThreadType.Conversation, // type
+        undefined, // entityType
+        undefined, // fqn
+        undefined, // taskStatus
+        8 // limit to 8 items for widget
+      );
+    }
+  }, [currentUser, getFeedData, selectedFilter]);
+
+  // Filter options for the dropdown
+  const filterOptions = [
+    {
+      label: t('label.all-activity'),
+      value: FeedFilter.ALL,
+      key: FeedFilter.ALL,
+    },
+    {
+      label: t('label.my-data'),
+      value: FeedFilter.OWNER,
+      key: FeedFilter.OWNER,
+    },
+    {
+      label: t('label.following'),
+      value: FeedFilter.FOLLOWS,
+      key: FeedFilter.FOLLOWS,
+    },
+  ];
+
+  const handleFilterChange = useCallback(({ key }: { key: string }) => {
+    setSelectedFilter(key as FeedFilter);
+  }, []);
+
+  const widgetIcon = useMemo(() => {
+    return customizeMyDataPageClassBase.getWidgetIconFromKey(widgetKey);
+  }, [widgetKey]);
+
+  const handleCloseClick = useCallback(() => {
+    !isUndefined(handleRemoveWidget) && handleRemoveWidget(widgetKey);
+  }, [widgetKey]);
+
+  const handleUpdateEntityDetails = useCallback(() => {
+    // Refresh feed data after update
+    if (getFeedData) {
+      getFeedData();
+    }
+  }, [getFeedData]);
+
+  const handleSizeChange = useCallback(
+    (value: number) => {
+      if (handleLayoutUpdate) {
+        const hasCurrentWidget = currentLayout?.find(
+          (layout: WidgetConfig) => layout.i === widgetKey
+        );
+
+        const updatedLayout = hasCurrentWidget
+          ? currentLayout?.map((layout: WidgetConfig) =>
+              layout.i === widgetKey ? { ...layout, w: value } : layout
+            )
+          : [
+              ...(currentLayout || []),
+              {
+                ...customizeMyDataPageClassBase.defaultLayout.find(
+                  (layout: WidgetConfig) => layout.i === widgetKey
+                ),
+                i: widgetKey,
+                w: value,
+              },
+            ];
+
+        handleLayoutUpdate(updatedLayout as Layout[]);
+      }
+    },
+    [currentLayout, handleLayoutUpdate, widgetKey]
+  );
+
+  const handleMoreClick = (e: MenuInfo) => {
+    if (e.key === WIDGETS_MORE_MENU_KEYS.REMOVE_WIDGET) {
+      handleCloseClick();
+    } else if (e.key === WIDGETS_MORE_MENU_KEYS.HALF_SIZE) {
+      handleSizeChange(1);
+    } else if (e.key === WIDGETS_MORE_MENU_KEYS.FULL_SIZE) {
+      handleSizeChange(2);
+    }
+  };
+
+  return (
+    <Card
```

Comment:
Use the common widget wrapper; everything below is handled in it. For reference, check the Data assets widget.